recipes/conversation_view/message_input/message_input.test.js

```diff
@@ -0,0 +1,241 @@
+import { mount } from '@vue/test-utils';
+import DtRecipeMessageInput from './message_input.vue';
+import { beforeEach, describe } from '@/node_modules/vitest/dist/index';
+
+import {
+  itBehavesLikeEmitsExpectedEvent,
+} from '../../../tests/shared_examples/events';
+
+// Wrappers
+let wrapper;
+let editor;
+// let editorEl;
+let imageInputEl;
+let messageInputEl;
+let characterLimitEl;
+let errorNoticeEl;
+
+let imageBtn;
+let emojiPickerBtn;
+let sendBtn;
+
+// let emojiPickerPopover;
+
+let footerLeftSlot;
+let footerRightSlot;
+
+// Test Environment
+let props;
+let attrs;
+let slots;
+let listeners;
+let imageUploadStub;
+
+// Constants
+const baseProps = {
+  modelValue: 'initial value',
+  inputAriaLabel: 'aria-label text',
+  inputClass: 'qa-editor',
+};
+
+const randoText = 'ricketyrick';
+
+const baseSlots = {
+  middle: 'image carousel',
+  footerLeft: 'footer left',
+  footerRight: 'footer right',
+};
+
+// Helpers
+const _setChildWrappers = () => {
+  editor = wrapper.find('[data-qa="dt-rich-text-editor"]').find('div[contenteditable]');
+
+  // buttons
+  imageBtn = wrapper.find('[data-qa="dt-message-input-image-btn"]');
+  emojiPickerBtn = wrapper.find('[data-qa="dt-message-input-emoji-picker-btn"]');
+  sendBtn = wrapper.find('[data-qa="dt-message-input-send-btn"]');
+
+  // Els
+  imageInputEl = wrapper.find('[data-qa="dt-message-input-image-input"]');
+  messageInputEl = wrapper.find('[data-qa="dt-message-input"]');
+  characterLimitEl = wrapper.find('[data-qa="dt-message-input-character-limit"]');
+  errorNoticeEl = wrapper.find('[data-qa="dt-message-input-error-notice"]');
+
+  // slot divs
+  footerLeftSlot = wrapper.find('[data-qa="dt-message-input-footer-left"]');
+  footerRightSlot = wrapper.find('[data-qa="dt-message-input-footer-right"]');
+};
+
+const _mountWrapper = () => {
+  wrapper = mount(DtRecipeMessageInput, {
+    props,
+    listeners,
+    attrs,
+    slots,
+    attachTo: document.body,
+  });
+};
+
+describe('DtMessage tests', () => {
+  // Test Setup
+  beforeAll(() => {
+    global.Range.prototype.getClientRects = vi.fn(() => [{}]);
+    global.Range.prototype.getBoundingClientRect = vi.fn(() => [{}]);
+    global.scrollBy = vi.fn();
+  });
+
+  beforeEach(async () => {
+    props = baseProps;
+    imageUploadStub = vi.fn();
```

Comment:
Here you can create
const imageUploadStub = vi.fn();

Reply:
not needed, left over code deleted. thanks!